### ISSUE TITLE
Geodesic module displays broken character

### ISSUE THREAD

**Issue by osulli** [cloudposse/geodesic#417]
## what

- Broken character in CLI.

## why

- Character should be escaped across all platforms.

## Versions

```
FROM cloudposse/terraform-root-modules:0.58.1 as terraform-root-modules

FROM cloudposse/helmfiles:0.22.0 as helmfiles

FROM cloudposse/geodesic:0.77.2
```

```
$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 18.10
Release:	18.10
Codename:	cosmic
```

## Screenshot

![image](https://user-images.githubusercontent.com/46930728/54748455-99a8d880-4bc9-11e9-80df-3cad4aa429e6.png)

```
 ✓   (healthera-sandbox-iac) aws ⨠ 
```

**Comment by osterman** [comment]
@osulli can you explain why the markdown looks good with the `⨠` and the screenshot looks bad?

**Comment by osterman** [comment]
basically it comes down to us using an unsupported UTF8 character for your terminal. we can update our default, since you're not the first person to have the problem.

**Comment by osulli** [comment]
That's interesting. I didn't notice it working in the markdown. 
I'm on my phone now and it looks ok, I wonder if it just doesn't work on ubuntu entirely.

**Comment by osulli** [comment]
From my Ubuntu 18 laptop:
![image](https://user-images.githubusercontent.com/46930728/55175887-10b91080-5178-11e9-8612-202aa3cb74ba.png)

It's clearly bigger than the terminal or Geodesic and clearly just a 'missing character' from ubuntu.

**Comment by osulli** [comment]
![image](https://user-images.githubusercontent.com/46930728/55175940-23cbe080-5178-11e9-9ee9-38ee34fda8f7.png)

**Comment by tamsky** [comment]
> 'missing character' from ubuntu.
To be clear, this is not an ubuntu-specific issue.
It's a missing glyph in the specific font or fonts being used in terminals or browsers.

Viewing this github comment in my browser, it's visible:
https://github.com/cloudposse/geodesic/issues/417#issuecomment-475809612

![image](https://user-images.githubusercontent.com/96978/55178481-b5b8f680-5142-11e9-86df-45448c1cb797.png)

**Comment by tamsky** [comment]
@osulli 
what terminal program is in the screenshot?
are you able to determine what font your terminal is using?
have you tried any other fonts?

**Comment by Nuru** [comment]
@osulli One option, if it works for you, is to set `export PROMPT_STYLE=unicode` in your [preferences file](https://github.com/cloudposse/geodesic/pull/422). Those are the nicest glyphs, but are not the default because they are "ambiguous width" Unicode characters that cause serious issues with command line editing on some terminals.  You can also use "plain" instead of "unicode" to get basic ASCII characters, but you may find that uglier than the broken Unicode glyph.

**Comment by Nuru** [comment]
The broken glyph '⨠' is Z NOTATION SCHEMA PIPING from the "Supplemental Mathematical Operators" Unicode block. As you can infer from the word "Supplemental" it appears to be not as widely supported as some of the other code blocks. I like it (when it is supported) because it is big compared to the alternatives. More importantly, according to [Ubuntu fonts](https://design.ubuntu.com/font/) it is a supported character on Ubutnu. So the real question @osulli is what font are you using in the terminal. You can use "Ubutnu monospace" which supports the character, or of course you can download any of a number of free fonts that support it, such as Google's [Noto Mono](https://www.google.com/get/noto/#mono-mono) or [GNU Free Font Mono](https://www.gnu.org/software/freefont/index.html).  See [this article](https://itsfoss.com/install-fonts-ubuntu-1404-1410/) about how to install fonts (and see what you already have installed) on Ubutntu.

Alternative glyphs include  
- » ($'\u00bb') RIGHT-POINTING DOUBLE ANGLE QUOTATION MARK most likely the most widely supported because it s from the Latin-1 supplement, but also the smallest.
- ≫ ($'\u226b') MUCH GREATER-THAN and
- ⋙ ($'\u22d9') VERY MUCH GREATER-THAN which are from the Mathematical Operators code block and therefore I would expect are more widely supported. Ubuntu is supposed to have them.
- ➤ ($'\u27a4') BLACK RIGHTWARDS ARROWHEAD from the Dingbats block, and which is used in the "unicode" style of prompt
- &#x25b6;&#xFE0E; ($'\u25b6') BLACK RIGHT-POINTING TRIANGLE which is sometimes presented as an emoji (as GitHub likes to) &#x25b6;&#xFE0F; 


Geodesic does not currently support customizing the glyph by itself, but as I mentioned above, it supports 2 alternatives to the default, `export PROMPT_STYLE=unicode` and export PROMPT_STYLE=plain`, that you can try and see if you like better. If anyone has a better suggestion for the prompt glyph to use, please add it to the comments below.

**Comment by osulli** [comment]
> @osulli
> what terminal program is in the screenshot?
> are you able to determine what font your terminal is using?
> have you tried any other fonts?

Standard ubuntu terminal.

I was on "Monogspace Regular". I tried "Ubuntu Mono Regular" which also did not solve the glyph issue.
![image](https://user-images.githubusercontent.com/46930728/55223614-7f44af80-5206-11e9-91ce-3a42d7684a34.png)

**Comment by osulli** [comment]
@Nuru 
Happy to use the unicode setting.

As per https://github.com/cloudposse/geodesic/pull/422 I figured on Geodesic 0.85.0 executing:
`echo $GEODESIC_DOT_DIR`  would return /localhost/.geodesic, however it is not set.
The file is present and contains:
```
export PROMPT_STYLE=unicode
```

I also added and rebuilt my module with this in the Dockerfile but no luck:
```
# Geodesic customization
ENV GEODESIC_DOT_DIR=/localhost/.geodesic
```

Have I missed a setup instruction to use the geodesic dot file?

**Comment by Nuru** [comment]
@osulli It is odd that your terminal still has this issue with the Ubuntu font. What is your locale set to? In particular, check `LANG`, `LC_ALL`, and `LC_CTYPE`.  Try `export LC_CTYPE="en_US.UTF-8"` in your terminal before starting up the Geodesic shell and again inside the shell and see if that helps.

My apologies about your difficulty setting up your preferences. Reviewers did not like the name `GEODESIC_DOT_DIR` and so it got changed to `GEODESIC_CONFIG_HOME` in the code but the documentation did not get updated (until now). However, the default is still `/localhost/.geodesic`, so `/localhost/.geodesic/preferences` will get loaded even if you did not set any environment variable (unless there is a directory called `/localhost/.geodesic/defaults` in which case  `/localhost/.geodesic/preferences` would be ignored and `/localhost/.geodesic/defaults/preferences` would be loaded instead.)

In any case, setting it in your preferences is just a convenience so that it automatically set up the way you like. You can test out the setting by running `export PROMPT_STYLE=unicode` on the `bash` command line inside Geodesic and see the results immediately.

**Comment by Nuru** [comment]
With release [0.86.2](https://github.com/cloudposse/geodesic/releases/tag/0.86.2) we added a `GEODISIC_PROMPT_GLYPHS` environment variable that you can set to set your own glyphs to show up at the end of the prompt. Keep in mind you will probably want to include a space at the end.

**Comment by osulli** [comment]
In my terminal:
```
echo $LANG
en_GB.UTF-8
```

In my module:
```
echo $LANG $LC_ALL $LC_CTYPE
en_US.UTF-8 en_US.UTF-8
```

**Comment by osulli** [comment]
```
oscar@infinitum:~$ pwd
/home/oscar
oscar@infinitum:~$ cat .geodesic/preferences 
export PROMPT_STYLE=unicode
```

Thank you this works fine now.

**Comment by osulli** [comment]
@Nuru please can you give an example on how to use `GEODISIC_PROMPT_GLYPHS`?

**Comment by Nuru** [comment]
@osulli Just set `GEODISIC_PROMPT_GLYPHS` to whatever string of characters you want at the end of the command line prompt. 

```
export GEODISIC_PROMPT_GLYPHS=$'\u00bb '
```

**Comment by Nuru** [comment]
Considering this closed by #429, since we cannot auto-detect what characters are available. If someone has a better solution, please open a new issue or pull-request describing or implementing it.

**Comment by Nuru** [comment]
Our updated recommendation is to install the `noto` font subset that includes the missing character
```
$ sudo apt update
$ sudo apt install fonts-noto-core
```

Alternatively, #756 introduces new options for changing the character, making our fallback recommendation setting `export PROMPT_STYLE=fancy` in your customizations. See the PR for details.
